### PR TITLE
Use fake iframe element to check reload result and replace on success.

### LINKED ISSUE #8728: Live reload shows offline page instead of reloading the page on slower machine
This is a timing issue: When you do not have fast enough computer, this keeps occurring regularly.

For example, on my 4-core 3.3GhZ i5 iMac with 16GB memory and 4TB SSD, route is lost every time one makes a code change to a view. Thus Spring DevTools and reload is completely broken on anything but the latest computers. 

My environment where bug can be reproduced reliably:
- MacOS 10.15.5 (19F101)
- iMac (Retina 5K, 27-inch, Mid 2015), 3.3 GHz Quad-Core Intel Core i5, 16 GB 1600 MHz DDR3
- OpenJDK Runtime Environment (build 14.0.1+14)
- Node v14.5.0
- Apache Maven 3.6.3
- Chrome 83.0.4103.116 (Official Build) (64-bit)
- Vaadin 14.3.0

To reproduce
- Download default app starter from https://vaadin.com/start/v14 (Spring Boot)
- Unzip
- `mvn`
- Open localhost:8080 in Chrome and see that app runs correctly and shows that live reload is available with Spring Boot Devtools and is enabled
- In MainView.java, add `add(new Button("Foo"));` to end of `MainView` method.
- `mvn compile`
- **_Browser incorrectly updates to shows "You are offline" page for "Vaadin Application"_**

To verify that the server-side change was actually done correctly
- Reload page
- Page loads correctly

What should have happened
- Live reloading should have loaded page only after server update has happened
- When server connection has been lost, we could show "waiting for server" message or indicator in bottom right corner live reload icon

### PR DESCRIPTION
**Pull request by denis-anisimov**
Avoid reload() method whose usage may lead to showing a blank page
because server is not yet restarted. Instead use a fake iframe element
inside gizmo which refers to the same URL. Its window may be reloaded
safely and the result can be checked before applying it to the main
document. If reload was successful (page is initialized with Vaadin JS)
then replace the document content  with the content of iframe.

Fixes #8728

### CODE REVIEW

**Comment by denis-anisimov** [comment]
The code works only for modern CCDM bootstraping mode.
It won't work with V14 style. 
This mode needs a separate code.

File: flow-client/src/main/resources/META-INF/resources/frontend/VaadinDevmodeGizmo.js [hunk 1]
```diff
@@ -974,7 +1029,7 @@ class VaadinDevmodeGizmo extends LitElement {
               <input id="toggle" type="checkbox"
                   ?disabled=${this.status === VaadinDevmodeGizmo.UNAVAILABLE || this.status === VaadinDevmodeGizmo.ERROR}
                   ?checked="${this.status === VaadinDevmodeGizmo.ACTIVE}"
-                  @change=${e => this.setActive(e.target.checked)}/>
+                  @ch   ange=${e => this.setActive(e.target.checked)}/>
```
Comment (joheriks) [inline flow-client/src/main/resources/META-INF/resources/frontend/VaadinDevmodeGizmo.js:1032]: This looks wrong.

File: flow-client/src/main/resources/META-INF/resources/frontend/VaadinDevmodeGizmo.js [hunk 2]
```diff
@@ -551,7 +551,8 @@ class VaadinDevmodeGizmo extends LitElement {
       status: {type: String},
       reloadConnectionBaseUri: {type: String},
       liveReloadBackend: {type: String},
-      springBootDevToolsPort: {type: Number}
+      springBootDevToolsPort: {type: Number},
+      reloadURL: {type: String}
```
Comment (joheriks) [inline flow-client/src/main/resources/META-INF/resources/frontend/VaadinDevmodeGizmo.js:555]: Not used?

File: flow-client/src/main/resources/META-INF/resources/frontend/VaadinDevmodeGizmo.js [hunk 3]
```diff
@@ -1004,7 +1059,8 @@ class VaadinDevmodeGizmo extends LitElement {
       ? html`<span class="status-description">${this.splashMessage}</span></div>`
       : html`<span class="status-description">Live reload ${this.status} </span><span class="ahreflike">Details</span></div>`
     }
-    </div>`;
+    </div>
+    <iframe style='display: none;' width='0px' height='0px' src='${window.location.href}' id='reload-frame'"></iframe>`;
```
Comment (joheriks) [inline flow-client/src/main/resources/META-INF/resources/frontend/VaadinDevmodeGizmo.js:1063]: Looks like superflous `"` there at position 107.

Comment (denis-anisimov) [reply flow-client/src/main/resources/META-INF/resources/frontend/VaadinDevmodeGizmo.js]: Thanks.

**Pull request merged**